ghost/email-service/lib/EmailEventStorage.js

```diff
@@ -128,11 +139,24 @@ class EmailEventStorage {
         }
     }
 
-    async unsubscribeFromNewsletters(event) {
+    async findNewslettersToKeep(event) {
         try {
-            await this.#membersRepository.update({newsletters: []}, {id: event.memberId});
+            const member = await this.#membersRepository.get({email: event.email}, {
+                withRelated: ['newsletters']
+            });
+            const existingNewsletters = member.related('newsletters');
+
+            if (!existingNewsletters) {
+                return [];
+            }
+
+            const email = await this.#models.Email.findOne({id: event.emailId});
+            const newsletterToRemove = await email.get('newsletter_id');
```

Comment:
```suggestion
            const newsletterToRemove = email.get('newsletter_id');
```

Reply:
thanks! Updated in 35f0ab52b0bdc6db3da4e82e92a5f91338686728